powerauth-push-server/src/main/java/io/getlime/push/controller/rest/CampaignController.java

```diff
@@ -0,0 +1,297 @@
+import io.getlime.push.model.entity.PushMessage;
+import io.getlime.push.model.entity.PushMessageBody;
+import io.getlime.push.model.request.CreateCampaignRequest;
+import io.getlime.push.model.request.DeleteCampaignRequest;
+import io.getlime.push.model.request.TestingCampaignRequest;
+import io.getlime.push.model.response.CreateCampaignResponse;
+import io.getlime.push.model.response.CampaignResponse;
+import io.getlime.push.model.response.ListOfCampaignResponse;
+import io.getlime.push.model.response.ListOfUsersFromCampaignResponse;
+import io.getlime.push.repository.PushCampaignDevicesRepository;
+import io.getlime.push.repository.PushCampaignRepository;
+import io.getlime.push.repository.PushCampaignUsersRepository;
+import io.getlime.push.repository.model.PushCampaign;
+import io.getlime.push.repository.model.PushCampaignUser;
+import io.getlime.push.service.PushSenderService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Controller;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Controller for push campaign related methods
+ *
+ * @author Martin Tupy, martin.tupy.work@gmail.com
+ */
+
+@Controller
+@RequestMapping(value = "push/campaign")
+public class CampaignController {
+
+    private PushCampaignRepository pushCampaignRepository;
+    private PushCampaignUsersRepository pushCampaignUsersRepository;
+    private PushCampaignDevicesRepository pushCampaignDevicesRepository;
+    private PushSenderService pushSenderService;
+
+    @Autowired
+    public CampaignController(PushCampaignRepository pushCampaignRepository, PushCampaignUsersRepository pushCampaignUsersRepository, PushSenderService pushSenderService) {
+        this.pushCampaignRepository = pushCampaignRepository;
+        this.pushCampaignUsersRepository = pushCampaignUsersRepository;
+        this.pushSenderService = pushSenderService;
+    }
+
+
+    /**
+     * Create a campaign for specific app.
+     * @param request id of specific app, body of specific messageBody
+     * @return ID for created campaign.
+     */
+    @RequestMapping(value = "create", method = RequestMethod.POST)
+    @ResponseBody
+    public ObjectResponse<CreateCampaignResponse> createCampaign(@RequestBody ObjectRequest<CreateCampaignRequest> request) {
+
+        CreateCampaignRequest requestObject = request.getRequestObject();
+
+        PushCampaign campaign = new PushCampaign();
+        campaign.setAppId(requestObject.getAppId());
+        campaign.setSent(false);
+        campaign.setTimestampCreated(new Date());
+
+        PushMessageBody message = requestObject.getMessage();
+
+        String messageString = null;
```

Comment:
Maybe we should replace this block with a private method `serializePushMessage(pmb)`, just as we do with the opposite procedure.